CONTRIBUTING.md

```diff
@@ -110,3 +105,68 @@ You need to be in the directory in which the images are defined and have a `resi
 find . -name '*-gray.svg' -type f -exec inkscape {} -w 72 -h 72 --export-filename resize/{}.png \; && find ./resize/ -type f -name '*.svg.png' -exec sh -c 'f="{}"; mv -- "$f" "${f%.svg.png}.png"' \;
 ```
 
+### Code Architecture
+
+In this section we'll explain the main packages if you want to contribute/read/understand how InfraMap does what it does.
+
+#### Providers
+
+The main idea behind InfraMap is to abstract each specific provider logic into a different implementation with a common interface
+so then we can have a common logic that calls this interface without knowing which provider is using. All this interface implementations
+are inside the `provider/` package and being `provider.Provider` the main interface with all the needed methods for each provider.
+
+Not all the methods are needed to be implemented and that's why we have the `provider/nop_provider.go` which has an empty implementation
+of the `provider.Provider` interface.
+
+All the providers that we have implemented are defined on `provider/{provider-name}` for example `provider/aws/` which holds
+the implementation of the AWS provider.
+
+For more information on the provider interface we recommend to read the interface itself.
```

Comment:
We could add a doc link: https://pkg.go.dev/github.com/cycloidio/inframap@v0.3.0/provider?tab=doc#Provider